lib/sshex.ex

```diff
@@ -198,65 +207,110 @@ defmodule SSHEx do
-
     # if we got status and closed, then we are done
-    parsed = case {status, closed} do
-      {st, true} when not is_nil(st) -> format_response({:ok, stdout, stderr, status}, opts)
-      _ -> receive_and_parse_response(conn, channel, opts[:connection_module],
-                                      timeout, stdout, stderr, status, closed)
-    end
+    parsed =
+      case {status, closed} do
+        {st, true} when not is_nil(st) ->
+          format_response({:ok, stdout, stderr, status}, opts)
+
+        _ ->
+          receive_and_parse_response(
+            conn,
+            channel,
+            opts[:connection_module],
+            timeout,
+            stdout,
+            stderr,
+            status,
+            closed
+          )
+      end
 
     # tail recursion
     case parsed do
-      {:loop, {ch, tout, out, err, st, cl}} -> # loop again, still things missing
+      # loop again, still things missing
+      {:loop, {ch, tout, out, err, st, cl}} ->
         get_response(conn, ch, tout, out, err, st, cl, opts)
-      x -> x
+
+      x ->
+        x
     end
   end
 
   # Parse ugly response
   #
-  defp receive_and_parse_response(conn, chn, connection_module, tout,
-                                  stdout \\ "", stderr \\ "", status \\ nil, closed \\ false) do
-    response = receive do
-      {:ssh_cm, ^conn, res} -> res
-    after
-      tout -> {:error, "Timeout. Did not receive data for #{tout}ms."}
-    end
+  defp receive_and_parse_response(
+         conn,
+         chn,
+         connection_module,
+         tout,
+         stdout \\ "",
+         stderr \\ "",
+         status \\ nil,
+         closed \\ false
+       ) do
+    response =
+      receive do
+        {:ssh_cm, ^conn, res} -> res
+      after
+        tout -> {:error, "Timeout. Did not receive data for #{tout}ms."}
+      end
 
     # call adjust_window to allow more data income, but only when needed
     case response do
-      {:data, ^chn, _, new_data} -> connection_module.adjust_window(conn, chn, byte_size(new_data))
-      _ -> :ok
+      {:data, ^chn, _, new_data} ->
+        connection_module.adjust_window(conn, chn, byte_size(new_data))
+
+      _ ->
+        :ok
     end
 
     case response do
-      {:data, ^chn, 1, new_data} ->       {:loop, {chn, tout, stdout, stderr <> new_data, status, closed}}
-      {:data, ^chn, 0, new_data} ->       {:loop, {chn, tout, stdout <> new_data, stderr, status, closed}}
-      {:eof, ^chn} ->                     {:loop, {chn, tout, stdout, stderr, status, closed}}
-      {:exit_signal, ^chn, _, _} ->       {:loop, {chn, tout, stdout, stderr, status, closed}}
-      {:exit_signal, ^chn, _, _, _} ->    {:loop, {chn, tout, stdout, stderr, status, closed}}
-      {:exit_status, ^chn, new_status} -> {:loop, {chn, tout, stdout, stderr, new_status, closed}}
-      {:closed, ^chn} ->                  {:loop, {chn, tout, stdout, stderr, status, true}}
-      any -> any # {:error, reason}
```

Comment:
FYI This particular indentation in 241-248 was very useful to get people to quickly understand this particular response mapping. It makes differences outstanding.

But this is your baby now! :rocket:
Just pointing out that I had to clarify this part to several devs back in the day, so it may be useful to you too.

Reply:
@rubencaro To be honest, I just ran `mix format` for format the whole code-base. None of the edits were manual (except bumping versions). I can clearly see what you mean know with having the lines neatly aligned.

Great feedback. Thank you :sun_with_face:

Reply:
@rubencaro I've tried to find ways to keep the formatting without "hacking" to much. I think it is very hard to "skip" the formatter for a certain block of code.  

In the end I think we should just go with the default formatting. I can see your point in having it "compact". But I think the default is good enough. But it is not an easy call. And thank you once again for pointing out this key part of the code.